### PR TITLE
Upgrade to Babel 7

### PR DESCRIPTION
**Pull request by vlad-zhukov**
Upgraded tests automatically, we need to go through them to make sure they work properly.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  "presets": ["react", "es2015", "stage-0"]
+  "presets": ["es2015", "react", "flow", "stage-0"]
```
Comment (vlad-zhukov) [inline .babelrc]: The 'flow' preset is required now because it was removed from the 'react' preset.

File: .eslintrc.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ module.exports = {
     'flowtype',
   ],
   rules: {
-    'array-bracket-spacing': 'off', // use babel plugin rule
+    'array-bracket-spacing': ['error', 'never'],
```
Comment (vlad-zhukov) [inline .eslintrc.js]: The 'babel/array-bracket-spacing' rule is deprecated and suggest to use 'array-bracket-spacing'

File: src/index.js [hunk 3]
```diff
@@ -90,7 +90,13 @@ export default function ({ template, types, traverse }) {
               };
             }
 
-            const plugin = require(pluginName);
+            let plugin = require(pluginName);
+
+            // Required for `babel-plugin-transform-flow-strip-types`
+            if (typeof plugin !== 'function') {
+              plugin = plugin.default;
+            }
```
Comment (vlad-zhukov) [inline src/index.js]: The 'babel-plugin-transform-flow-strip-types' exports default

File: test/fixtures/es-class-extend-deopt/expected-remove-es6.js [hunk 4]
```diff
@@ -2,7 +2,14 @@ import BaseComponent from 'components/base';
...
 class Foo extends BaseComponent {
   render() {}
+
 }
-Foo.propTypes = {
-  foo: PropTypes.string.isRequired
-};
\ No newline at end of file
+
+Object.defineProperty(Foo, "propTypes", {
```
Comment (vlad-zhukov) [inline test/fixtures/es-class-extend-deopt/expected-remove-es6.js]: Broken test? Shouldn't it remove it?

Comment (oliviertassinari) [reply test/fixtures/es-class-extend-deopt/expected-remove-es6.js]: How is this broken?

Comment (vlad-zhukov) [reply test/fixtures/es-class-extend-deopt/expected-remove-es6.js]: It should remove prop types, isn't it? Am I not noticing something obvious?

Comment (oliviertassinari) [reply test/fixtures/es-class-extend-deopt/expected-remove-es6.js]: It shouldn't. It's here to show a limitation of the lib. It's a tradeoff taken for perf.

Comment (vlad-zhukov) [reply test/fixtures/es-class-extend-deopt/expected-remove-es6.js]: I see, that's why it's not working for me as I expect it to. Looks like a bug to be tho.

Comment (oliviertassinari) [reply test/fixtures/es-class-extend-deopt/expected-remove-es6.js]: @vlad-zhukov It's related to #8

**Comment by vlad-zhukov** [comment]
I've published this branch at `@vlad-zhukov/babel-plugin-transform-react-remove-prop-types` and tested on my project against the currently published one. The output is exactly the same (same content hashes).

**Comment by oliviertassinari** [comment]
@vlad-zhukov Let's release it :)

**Comment by oliviertassinari** [comment]
@vlad-zhukov Thanks!

**Comment by vlad-zhukov** [comment]
Thank you!

**Pull request merged**